lib/Client.php

```diff
@@ -117,6 +117,12 @@ public function read($size = null) {
         }
 
         $state = $this->state;
+        if (isset($state->buffer[0]) && ($size === null || isset($state->buffer[$size-1]))) {
```

Comment:
This is emptying the buffer if there are multiple concurrent read() calls with different sizes (first call a higher size than the second one); thus extend that condition by `!$state->readOperations && `, giving:

```php
if (!$state->readOperations && isset($state->buffer[$size === null ? 0 : $size - 1]))
```